src/signal.rs

```diff
@@ -75,10 +79,18 @@ impl Display for Error {
     }
 }
 
+/// A specialized [Result](https://doc.rust-lang.org/std/result/enum.Result.html) type
+/// for operations that can return [`Error`]: Enum.error.html.
+///
+/// This typedef is generally used to avoid writing out `Error` directly and is
+/// otherwise a direct mapping to `Result`.
 pub type SignalResult<T> = result::Result<T, Error>;
 type SiginfoHandler = extern "C" fn(num: c_int, info: *mut siginfo_t, _unused: *mut c_void) -> ();
 
+/// Enum of signal handlers for
+/// [`sigaction`](http://man7.org/linux/man-pages/man2/sigaction.2.html).
 pub enum SignalHandler {
+    /// Specify the signal handler `sa_sigaction` of sigaction structure.
     Siginfo(SiginfoHandler),
     // TODO add a`SimpleHandler` when `libc` adds `sa_handler` support to `sigaction`.
```

Comment:
I don't know if this is even relevant anymore since the recommended way to handle signals is `sigaction` not `signal`/ I'd remove the comment. And maybe consider removing the `enum` altogether since there'd only be a single variant left.

Reply:
I think the enum means there reserves a space for `sa_handler` of `sigaction` structure.

Reply:
I think enum means reserving a space for `sa_handler` of `sigaction` structure in future.

Reply:
Remove the "TODO" comment and keep enum for future.